tests/common/test_pkg_config.py

```diff
@@ -31,10 +49,27 @@ def test_load_default():
     check.equal(1, len(settings["QueryDefinitions"]["Default"]))
     for path in settings["QueryDefinitions"]["Default"]:
         check.is_true(type(path), str)
-        path = f"data/{path}"
-        check.is_true(
-            Path(pkg_config.__file__).resolve().parent.parent.joinpath(path).is_dir()
+        path = (
+            Path(pkg_config.__file__).resolve().parent.parent.joinpath(f"data/{path}")
         )
+        check.is_true(path.is_dir())
+        for query_file in path.rglob("*.yaml"):
+            validate_queries_file_structure(query_file)
+    for path in settings["QueryDefinitions"].get("Custom", []):
```

Comment:
I think it's maybe risky to be checking all the testdata files - since there may be some query files that are intentionally malformed.
The test is failing because it's getting "testdata" as the custom path from the msticpyconfig-test.yaml and one of the test files for the sentinel query downloaded is a query when the test code expects it not to be.
Seems like we only really want to test prod query files that are in the msticpy/data/queries folder. I think we can remove the second part of this test. Maybe can add something back here or build a special test with controlled msticpyconfig.yaml paths if we ever need to test query files in a custom folder.

Reply:
I understood why this test was failing. I did not define any of the "root keys" (ie: metadata, sources and defaults) to be required. Therefore, the Sentinel rules somehow matched the definition since it did not contain any of those keys...

I was also using this test to validate my own "custom queries" defined in my msticpyconfig file, but I do not know if that's a valid case?

Reply:
Hmm that seems valid. however there are or could be invalid query files in the tests\testdata folder.
I guess in your test you could remove the custom queries key from the config or use a custom msticpyconfig.
Maybe easier just to skip the path in your test if it points to the tests folder.

Reply:
Content from the test folder is now ignored from the package validation